lib/taglet/tag_as.ex

```diff
@@ -45,4 +49,32 @@ defmodule Taglet.TagAs do
       end
     end
   end
+
+  defmacro __before_compile__(_env) do
+
+    quote do
+      @contexts
+      |> Enum.each(fn(context) ->
+        Module.eval_quoted(__MODULE__, quote do
+          def unquote(:"add_#{context}")(tags) do
+            Taglet.add(%__MODULE__{}, tags, unquote(context))
+          end
+
+          def unquote(:"add_#{Inflex.singularize(context)}")(tags) do
```

Comment:
You can change `Inflex.singularize(context)` by `singularized_context`:https://github.com/bizneo/taglet/pull/4/files#diff-3ee991c1e4a4954b7ddbe5a4b2854690R4